src/main/java/io/github/Alathra/AlathranCommands/commands/CommandWildTp.java

```diff
@@ -11,14 +11,20 @@
 import io.github.Alathra.AlathranCommands.runnable.WildTeleportTask;
 import io.github.Alathra.AlathranCommands.utility.PlaytimeChecker;
 import io.github.Alathra.AlathranCommands.utility.TPCfg;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
 import org.bukkit.entity.Player;
 
 import java.time.Duration;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 public class CommandWildTp {
     private final int pricePreCooldown = TPCfg.get().getInt("Settings.WildTP.Price-pre");
     private final int pricePostCooldown = TPCfg.get().getInt("Settings.WildTP.Price-post");
+    private final boolean teleportCurrentWorld = TPCfg.get().getBoolean("Settings.WildTP.Teleport-current-world");
+    private final List<String> allowedWorlds = TPCfg.get().getStringList("Settings.WildTP.Allowed-origin-worlds");
+    private World teleportWorld;
```

Comment:
Odd to store the teleport world since the value of this variable is reassigned every time the command is run.

The field should be turned into a variable and declared at line ~36 as a `@Nullable World teleportWorld;`.